### DISCUSSION TITLE
NetConnection.Connect.Failed when I use RTMP stream with H264

### DISCUSSION THREAD

**Discussion by mrmda28** [HaishinKit/HaishinKit.swift#1487] [Q&A]
### Describe the bug

Since version 1.8.1, more specifically, after [this commit](https://github.com/shogo4405/HaishinKit.swift/commit/eda45afe95a9767830d2e4ccddc7b95d00a6442a), I connect to an rtmp stream and in rtmpStatusHandler comes the status NetConnection.Connect.Failed.
Using ffprobe my stream shows "codec_name": "h264" and "codec_long_name": "H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10".
I guess this is because of the line `public static let supportedFourCcList = ["hvc1"]` in the RTMPConnection class. If I add "h264" to this array, nothing changes. But if I remove the `"fourCcList": Self.supportedFourCcList,` line for the RTMPCommandMessage, everything will work as it should.

### To Reproduce

1. Connect to an RTMP stream with H264

### Expected behavior

RTMP stream with H264 is working correctly

### Version

1.8.1 - last

**Comment by shogo4405** [comment]
Yes, I understand that there are cases where some servers may not be able to connect, and I am making it public.

The fourCcList is specified in Enhanced RTMP. Naturally, older servers do not support it.
However, rather than not supporting Enhanced RTMP, I understand that servers may reject connections if they do not support the AMF Strict Array for RTMPCommandMessage.

What server are you using?

**Reply by mrmda28** [reply]
Thanks for the reply. 
I am using a video camera that creates an hotspot wifi. I connect to it and from that network I can access the rtmp stream from that camera. About the server I can't answer, I don't know.
Interesting feature, everything will work fine if supportedFourCcList = "" (any string) or supportedFourCcList: String? = nil.
Is it possible to add a booleans flag or some setting to not pass supportedFourCcList to "fourCcList"? Or is there some other way to do it in the code? What do you think is the best way to do it here?

**Reply by shogo4405** [reply]
I might consider an option to not send it, depending on the situation.

Can you tell me the product name of the video camera that creates the hotspot WiFi?
For example, if you set fourCcList to "test," is it also rejected by the server?

**Reply by mrmda28** [reply]
Unfortunately I can't tell you about this device because it is not yet available for sale.
If I set `"fourCcList": "test,"` it will work like with any other string. It will also work if I pass nil - `“fourCcList": nil`. But it won't work if I pass an array there.
